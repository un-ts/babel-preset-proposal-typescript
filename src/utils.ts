import compareVersions from 'compare-versions'

export const NODE_VERSION = process.versions.node

export const IS_RECORD_TUPLE_SUPPORTED = compareVersions.compare(
  NODE_VERSION,
  '14.6',
  '>=',
)
