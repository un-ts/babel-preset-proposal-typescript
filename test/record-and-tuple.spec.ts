import { IS_RECORD_TUPLE_SUPPORTED } from '../src/utils.js'

import { execute } from './helpers.js'

const proposal = 'record-and-tuple'

test(proposal, async () => {
  expect(() => execute(proposal)).toThrowErrorMatchingSnapshot()
  if (IS_RECORD_TUPLE_SUPPORTED) {
    const module = await import('./record-and-tuple.js')
    // eslint-disable-next-line jest/no-conditional-expect
    expect(module.default).toEqual(['Record & Tuple proposal', 'tc39'])
  }
})
