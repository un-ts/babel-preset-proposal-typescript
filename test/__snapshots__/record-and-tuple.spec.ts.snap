// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`record-and-tuple 1`] = `"Unexpected token ','"`;
